Finalize Raiden Mobile Wallet Android POC

## User Story

## Acceptance criteria

A minimal working Android version of the Light Client available on Google Play Store for internal download as well as documentation on how to publish the dApp for internal downloads for any user who would be interested in doing so for themselves. A user should be able to follow the documentation required for publishing their own branded Google Play store internal test version of the dApp.

## Tasks
- [ ] Test it and determine whether things are broken and needs to be fixed.

While deploying the fastlane actions 
`bundle exec fastlane android deploy_internal_test_draft`
I get this error
```
[18:10:31]: ▸ DONE  Build complete. The dist/bundled directory is ready to be deployed.
[18:10:31]: ▸ INFO  Check out deployment instructions at https://cli.vuejs.org/guide/deployment.html
[18:10:32]: ▸ ✔ Copying web assets from bundled to android/app/src/main/assets/public in 49.70ms
[18:10:32]: ▸ ✔ Creating capacitor.config.json in android/app/src/main/assets in 800.86μp
[18:10:32]: ▸ [info] Found 1 Cordova plugin for android:
[18:10:32]: ▸ cordova-sqlite-storage@6.0.0
[18:10:32]: ▸ ✔ copy android in 117.11ms
[18:10:32]: ▸ INFO  Building native app
[18:10:32]: ▸ FAILURE: Build failed with an exception.
[18:10:32]: ▸ * Where:
[18:10:32]: ▸ Settings file '/home/devaki/raidenforked/clone_light_client/light-client/raiden-dapp/android/settings.gradle' line: 5
[18:10:32]: ▸ * What went wrong:
[18:10:32]: ▸ A problem occurred evaluating settings 'android'.
[18:10:32]: ▸ > Could not read script '/home/devaki/raidenforked/clone_light_client/light-client/raiden-dapp/android/capacitor.settings.gradle' as it does not exist.
[18:10:32]: ▸ * Try:
[18:10:32]: ▸ Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
[18:10:32]: ▸ * Get more help at https://help.gradle.org
[18:10:32]: ▸ BUILD FAILED in 515ms
[18:10:33]: ▸ ERROR  Error: Command failed with exit code 1: ./gradlew assembleRelease
[18:10:33]: ▸ Error: Command failed with exit code 1: ./gradlew assembleRelease
[18:10:33]: ▸ at makeError (/home/devaki/raidenforked/clone_light_client/light-client/node_modules/vue-cli-plugin-capacitor/node_modules/execa/lib/error.js:59:11)
[18:10:33]: ▸ at handlePromise (/home/devaki/raidenforked/clone_light_client/light-client/node_modules/vue-cli-plugin-capacitor/node_modules/execa/index.js:114:26)
[18:10:33]: ▸ at runMicrotasks (<anonymous>)
[18:10:33]: ▸ at processTicksAndRejections (internal/process/task_queues.js:93:5)
[18:10:33]: ▸ at async /home/devaki/raidenforked/clone_light_client/light-client/node_modules/vue-cli-plugin-capacitor/index.js:60:11
[18:10:33]: ▸ error Command failed with exit code 1.
[18:10:33]: ▸ info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

This file does not exist 
Could not read script '/home/devaki/raidenforked/clone_light_client/light-client/raiden-dapp/android/capacitor.settings.gradle' as it does not exist.
**capacitor.settings.gradle**
For this I found we need to sync capacitor 
`npx cap sync android`
 and

> While deploying the fastlane actions
> `bundle exec fastlane android deploy_internal_test_draft`
> I get this error
> 
> ```
> [18:10:31]: ▸ DONE  Build complete. The dist/bundled directory is ready to be deployed.
> [18:10:31]: ▸ INFO  Check out deployment instructions at https://cli.vuejs.org/guide/deployment.html
> [18:10:32]: ▸ ✔ Copying web assets from bundled to android/app/src/main/assets/public in 49.70ms
> [18:10:32]: ▸ ✔ Creating capacitor.config.json in android/app/src/main/assets in 800.86μp
> [18:10:32]: ▸ [info] Found 1 Cordova plugin for android:
> [18:10:32]: ▸ cordova-sqlite-storage@6.0.0
> [18:10:32]: ▸ ✔ copy android in 117.11ms
> [18:10:32]: ▸ INFO  Building native app
> [18:10:32]: ▸ FAILURE: Build failed with an exception.
> [18:10:32]: ▸ * Where:
> [18:10:32]: ▸ Settings file '/home/devaki/raidenforked/clone_light_client/light-client/raiden-dapp/android/settings.gradle' line: 5
> [18:10:32]: ▸ * What went wrong:
> [18:10:32]: ▸ A problem occurred evaluating settings 'android'.
> [18:10:32]: ▸ > Could not read script '/home/devaki/raidenforked/clone_light_client/light-client/raiden-dapp/android/capacitor.settings.gradle' as it does not exist.
> [18:10:32]: ▸ * Try:
> [18:10:32]: ▸ Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
> [18:10:32]: ▸ * Get more help at https://help.gradle.org
> [18:10:32]: ▸ BUILD FAILED in 515ms
> [18:10:33]: ▸ ERROR  Error: Command failed with exit code 1: ./gradlew assembleRelease
> [18:10:33]: ▸ Error: Command failed with exit code 1: ./gradlew assembleRelease
> [18:10:33]: ▸ at makeError (/home/devaki/raidenforked/clone_light_client/light-client/node_modules/vue-cli-plugin-capacitor/node_modules/execa/lib/error.js:59:11)
> [18:10:33]: ▸ at handlePromise (/home/devaki/raidenforked/clone_light_client/light-client/node_modules/vue-cli-plugin-capacitor/node_modules/execa/index.js:114:26)
> [18:10:33]: ▸ at runMicrotasks (<anonymous>)
> [18:10:33]: ▸ at processTicksAndRejections (internal/process/task_queues.js:93:5)
> [18:10:33]: ▸ at async /home/devaki/raidenforked/clone_light_client/light-client/node_modules/vue-cli-plugin-capacitor/index.js:60:11
> [18:10:33]: ▸ error Command failed with exit code 1.
> [18:10:33]: ▸ info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
> ```
> 
> This file does not exist
> **Could not read script '/home/devaki/raidenforked/clone_light_client/light-client/raiden-dapp/android/capacitor.settings.gradle' as it does not exist.**
> **capacitor.settings.gradle**
> For this I found we need to sync capacitor
> `npx cap sync android`
> and then `bundle exec` runs smoothly

Good catch. I assume this wasn't necessart for us as we always used AndroidStudio in-between. I added it now. I also realized that the Capacitor config included an absolute path of my personal PC. I'm confused that it worked for you guys. Anyway I made it an relative path now.

--- a/raiden-dapp/README.md
+++ b/raiden-dapp/README.md
@@ -66,3 +66,9 @@ From the root directory run:
 ```bash
 yarn workspace raiden-dapp lint
 ```
+
+# Bundle & Publish to Mobil App Stores
+
+The bundling and publishing to the Google Play Store or the App Store by Apple
+is handled using _Fastlane_. Please refer to the [respective
+documentation](./fastlane/README.md) to get further details on that.

--- a/raiden-dapp/capacitor.config.json
+++ b/raiden-dapp/capacitor.config.json
@@ -1,1 +1,1 @@
-{"appId":"com.brainbot.raiden.dapp","appName":"Raiden dApp","bundledWebRuntime":false,"webDir":"/home/thore/projects/brainbot-projects/repositories/raiden/light-client-prs/raiden-dapp/dist/bundled","linuxAndroidStudioPath":"/usr/bin/android-studio"}
+{"appId":"com.brainbot.raiden.dapp","appName":"Raiden dApp","bundledWebRuntime":false,"webDir":"./dist/","linuxAndroidStudioPath":"/usr/bin/android-studio"}

--- a/raiden-dapp/fastlane/README.md
+++ b/raiden-dapp/fastlane/README.md
@@ -13,23 +13,25 @@ machines.
 
 ## Pre-requisites
 
-- Make sure _Ruby_ and its package manager `bundler` are available and executable
-  (`ruby --version` & `which bundler`). _Fastlane_ is written in _Ruby_ and can be
-  extended with plugins and other _RubyGems_.
+- Make sure _Ruby_ and its package manager `bundler` of version `2.4` [are
+  installed](https://www.ruby-lang.org/en/documentation/installation/) and
+  executable (`ruby --version` & `which bundler`). _Fastlane_ is written in
+  _Ruby_ and can be extended with plugins and other _RubyGems_.
 - For _Android_ releases, make sure that the _Java_ environment is installed like
   the _Gradle_ package manager (comes automatically if _Android Studio_ is
   installed).
 - For _iOS_ releases, make sure that the _XCode_ tool-box is installed (only
   works on Mac).
 - Make sure `yarn` is executable to build the Raiden Light Client dApp itself.
-- The first 3 steps of the dApp's [installation
+- The first few steps of the dApp's [installation
   instructions](https://github.com/raiden-network/light-client#install-and-run-the-dapp)
-  have been executed.
+  have been executed including the build of the SDK.
 
 
 ## Installation
 
-Install all _Ruby_ dependencies for _Fastlane_ and its plugins.
+Install all _Ruby_ dependencies for _Fastlane_ and its plugins. Execute this
+command inside the `raiden-dapp/` directory.
 
 ```sh
 $ bundler install

--- a/raiden-dapp/fastlane/README.md
+++ b/raiden-dapp/fastlane/README.md
@@ -46,6 +46,13 @@ Secret environment variables (e.g. passwords) are collected inside
 `./raiden-dapp/fastlane/.env.secrets`. The latter gets sourced by _Fastlane_
 itself automatically. It should not be sourced to the regular shell manually.
 
+**NOTE**
+The following instructions are targeting a release to the app store repositories
+owned by the Raiden project creators. Anyone else who tries to publish the dApp
+to their own repository need to adopt to use their own secrets. Anyway this
+should give the user a good idea of what secrets are necessary and where does
+the setup here expect to find them.
+
 **IMPORTANT**
 Secret data must **never** be checked into the version control system**!!** The
 locations used per default (as listed above) are set to be ignored. Please

--- a/raiden-dapp/package.json
+++ b/raiden-dapp/package.json
@@ -14,7 +14,7 @@
     "lint": "vue-cli-service lint --no-fix --max-warnings 0 && stylelint ./src/**/*.{vue,scss}",
     "lint:fix": "vue-cli-service lint --fix && stylelint --fix ./src/**/*.{vue,scss}",
     "capacitor:init:ios": "cap add ios",
-    "capacitor:init:android": "cap add android && sed --in-place 's/<application/<application android:usesCleartextTraffic=\"true\"/' ./android/app/src/main/AndroidManifest.xml",
+    "capacitor:init:android": "cap add android && cap sync android && sed --in-place 's/<application/<application android:usesCleartextTraffic=\"true\"/' ./android/app/src/main/AndroidManifest.xml",
     "capacitor:init": "f() { platform=${1##*--}; test -d ./$platform || yarn run capacitor:init:$platform; }; f",
     "capacitor:build": "f() { yarn capacitor:init $1 && vue-cli-service capacitor:build --mode capacitor $1; }; f",
     "capacitor:serve": "f() { yarn capacitor:init $1 && vue-cli-service capacitor:serve --mode capacitor $1; }; f",

